<?xml version="1.0"?>
<launch>
  <!-- <arg name="script" default="nav_cloning_node_pytorch_default.py"/> -->
  <!-- <arg name="script" default="nav_cloning_node_pytorch_rand_hsv.py"/> -->
  <!-- <arg name="script" default="nav_cloning_node_pytorch_rand_hsv_confirm.py"/> -->
  <arg name="script" default="nav_cloning_node_pytorch_rand_gamma.py"/>
  <!-- <arg name="script" default="nav_cloning_node_pytorch_rand_gamma_confirm.py"/> -->
  <!-- <arg name="script" default="nav_cloning_node_pytorch_mix.py"/> -->
  <!-- <arg name="script" default="nav_cloning_node_pytorch_action.py"/> -->
  <!-- <arg name="script" default="nav_cloning_node_pytorch_edge.py"/> -->
  <!-- <arg name="script" default="nav_cloning_node_pytorch_extreme.py"/> -->
  



  <!-- <arg name="mode" default="use_dl_output"/> -->
  <arg name="mode" default="change_dataset_balance"/>
  <!-- <arg name="mode" default="mix"/> -->
  <!-- <arg name="mode" default="extreme"/> -->
  <!-- <arg name="mode" default="action_extreme"/> -->
  <arg name="robot_x" default="2.915899" />
  <arg name="robot_y" default="0.717698" />
  <arg name="robot_Y" default="0.024363" />
  <rosparam param="/nav_cloning_node/mode" subst_value="True">$(arg mode)</rosparam>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find nav_cloning)/world/willow_garage.world"/> 
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <!-- <include file="$(find waypoint_server)/launch/waypoint_server.launch">
    <arg name="output" default="screen"/>
    <arg name="goal_topic" default="/move_base_simple/goal"/>
    <arg name="regist_goal_topic" default="/clicked_point"/>
    <arg name="config_file" default="$(find waypoint_server)/config/waypoint_server.yaml"/>
    <group ns="waypoint_manager">
      <node pkg="waypoint_server" type="waypoint_server_node" name="waypoint_server" output="$(arg output)">
        <rosparam command="load" file="$(arg config_file)" subst_value="true"/>
        <remap from="waypoint/regist_point" to="$(arg regist_goal_topic)"/>
      </node>
      <node pkg="waypoint_server" type="waypoint_to_posestamped_node" name="waypoint_to_posestamped" output="$(arg output)">
        <rosparam command="load" file="$(arg config_file)" subst_value="true"/>
        <remap from="move_base_simple/goal" to="$(arg goal_topic)"/>
      </node>
      <node pkg="waypoint_visualization" type="waypoint_visualization_node" name="waypoint_visualization" output="$(arg output)">
        <rosparam command="load" file="$(arg config_file)" subst_value="true"/>
        <remap from="waypoint/regist_point" to="$(arg regist_goal_topic)"/>
      </node>
      <node pkg="goal_event_handler" type="radius_node" name="radius_node" output="$(arg output)">
        <rosparam command="load" file="$(arg config_file)" subst_value="true"/>
      </node>
    </group>
  </include> -->
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_waffle_pi.urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model mobile_base -x $(arg robot_x) -y $(arg robot_y) -z 0.0 -Y $(arg robot_Y) -param robot_description" />
  <!-- <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model mobile_base -x 0.0 -y 0.0 -z 0.0 -param robot_description" /> -->

  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="5 nav_cloning nav_cloning.launch script:=$(arg script)"
        name="timed_roslaunch" output="screen" />

  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="8 nav_cloning turtlebot3_navigation.launch model:=waffle map_file:=$(find nav_cloning)/maps/willowgarage.yaml"
        name="timed_roslaunch2" >
  </node>
  <!-- <node pkg="timed_roslaunch" type="timed_roslaunch.sh" -->
  <!--       args="8 nav_cloning turtlebot3_navigation.launch model:=waffle map_file:=$(find nav_cloning)/maps/willowgarage.yaml waypoints_file:=$(find nav_cloning)/maps/test_loop.yaml" -->
  <!--       name="timed_roslaunch2" > -->
  <!-- </node> -->

  <!-- <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="20 nav_cloning start_wp_nav.launch"
        name="timed_roslaunch3" >
  </node> -->
</launch>
